src/jdk.jdwp.agent/share/native/libjdwp/threadControl.c

```diff
@@ -1086,8 +837,8 @@ resumeThreadByNode(ThreadNode *node)
     if (node->suspendCount > 0) {
         node->suspendCount--;
         debugMonitorNotifyAll(threadLock);
-        if ((node->suspendCount == 0) && node->toBeResumed &&
-            !node->suspendOnStart) {
+        if ((node->suspendCount == 0) && node->toBeResumed) {
+            JDI_ASSERT(!node->suspendOnStart);
```

Comment:
It'd be nice to add a small comment explaining why this asserted is here.

Reply:
Ok. The root reason is to sanity check that removing this check from the `if` statement is in fact ok to do. I guess that might be obvious to someone looking at the diff, but not someone just looking at the above code. I'll try to come up with a reasonable comment.